iceoryx2-pal/posix/src/linux/unistd.rs

```diff
@@ -17,9 +17,9 @@ use crate::posix::types::*;
 
 pub unsafe fn proc_pidpath(pid: pid_t, buffer: *mut c_char, buffer_len: size_t) -> isize {
     let path = if pid == crate::internal::getpid() {
-        "/proc/self/exe".to_owned()
+        "/proc/self/exe\0".to_owned()
```

Comment:
Forgot here that rust string literals are not 0 terminated ... Thankfully the address sanitizer on the C++ side found it.